### ISSUE TITLE
Csv reports and detecting charset encodings

### ISSUE THREAD

**Issue by gregordonsky** [jlevers/selling-partner-api#261]
After playing a bit downloading Amazon report documents, i noticed this line:

https://github.com/jlevers/selling-partner-api/blob/e30bf3c42774a2656a2c8ef7ac5b7ca391d3bbaa/lib/Document.php#L93

So i think that could be better to download the document with guzzle, and provide a better method for retrieving character encodings.

so, for example, using this code:
```php
$response = $this->client->request(
	'GET', $this->url, ['stream' => true]
);
	
$type = $response->getHeader('content-type');
$parsed = \GuzzleHttp\Psr7\Header::parse($type);
$rawContents = $response->getBody()->getContents();
```

I can get the rawContents using Guzzle.

In this variable:
```php
$parsed[0]['charset']
```

I have the charset.

Now it should be possibile to convert in UTF-8 the amazon report with this line:
```php
mb_convert_encoding($contents, 'UTF-8', $parsed[0]['charset']);
```

that automatically detects the charset of the downloaded resource.

What do you think about this  @jlevers ?

**Comment by jlevers** [comment]
Interesting, good idea! We'd need to have a check to make sure that the `content-type` header exists, and actually contains a `charset` key, and if either of those didn't exist choose a default encoding to assume.

Would you be interested in opening a PR to address this?

**Comment by gregordonsky** [comment]
> Would you be interested in opening a PR to address this?

Sure, but i don't open PR usually, the PR #269 is ok?

**Comment by jlevers** [comment]
Solved by #269.
